### PR TITLE
Fixes the TGUI Rebuild Action Again

### PR DESCRIPTION
**Pull request by MCterra10**
## About The Pull Request
I took zewaka's stuff, fixed their glaring oversight that they still don't understand and fixed the issue with changes getting into the commits from weird places.

Also changes the trigger command to be `?rebuild tgui` instead of `/rebuild tgui` so it's more in line with beebot.

## Why It's Good For The Repo
It actually works now lol

## Changelog
N/A

### CODE REVIEW

**Comment by jupyterkat** [comment]
make it runnable by pr authors pretty please

**Comment by MCterra10** [comment]
> make it runnable by pr authors pretty please

it literally is, just comment `?rebuild tgui` on your PR (if/when this is merged ofc)

**Comment by jupyterkat** [comment]
~I think only maints can run it rn~ nevermind

**Comment by PowerfulBacon** [comment]
What if something gets merged while its running, although unlikely?

**Comment by MCterra10** [comment]
> What if something gets merged while its running, although unlikely?

For the manual trigger one, it only commits to the pr, so nothing

For the automatic one that triggers on pr merge, it might break if you spam merge at just the right time, but it's set up to do a pull before committing and pushing, so should be fine

**Comment by ike709** [comment]
I'm not touching this with a 10 foot pole until Crossed approves it, but based.

File: .github/workflows/tgui_recompile.yml
Comment (Crossedfall) [inline .github/workflows/tgui_recompile.yml]: Use Beebot's information please.

File: .github/workflows/tgui_recompile.yml
Comment (Crossedfall) [inline .github/workflows/tgui_recompile.yml:91]: While I cannot currently think of a better way to handle this, I genuinely dislike nearly everything about this. It's incredibly dirty and (while somewhat minimal) poses potential security risks. I would far prefer a method that doesn't require a secondary build & commit after a PR is merged

**Review by Crossedfall — CHANGES_REQUESTED** [review]
email/name needs to be updated. 

Also, while I understand why it's there, I'm quite uncertain on how I feel on the auto_rebuild action compiling and pushing to master after a merge. This isn't a hard blocker, but I would honestly take any alternative.

File: .github/workflows/tgui_recompile.yml
Comment (MCterra10) [reply .github/workflows/tgui_recompile.yml:91]: I'm afraid that's not possible without performing a full merge of the master branch into the PR, as unless that happens, it's impossible to have an up-to-date build without author/maintainer intervention

Comment (MCterra10) [reply .github/workflows/tgui_recompile.yml:91]: Unless you want to trust the action to perform a full merge, that is

File: .github/workflows/tgui_recompile.yml
Comment (MCterra10) [reply .github/workflows/tgui_recompile.yml]: Will update

**Pull request merged**
